overlord/devicestate/handlers_gadget.go

```diff
@@ -105,15 +105,34 @@ func (m *DeviceManager) doUpdateGadgetAssets(t *state.Task, _ *tomb.Tomb) error
 		model = remodelCtx.Model()
 	}
 	// be extra paranoid when checking we are installing the right gadget
-	expectedGadgetSnap := model.Gadget()
-	if snapsup.InstanceName() != expectedGadgetSnap {
-		return fmt.Errorf("cannot apply gadget assets update from non-model gadget snap %q, expected %q snap",
-			snapsup.InstanceName(), expectedGadgetSnap)
-	}
+	var updateData *gadget.GadgetData
+	switch snapsup.Type {
+	case snap.TypeGadget:
```

Comment:
these changes also need unit tests changes?

Reply:
I added unit tests to gadget/devicestate now too that should cover these.